doc/rtd/topics/configuration.rst

```diff
@@ -0,0 +1,89 @@
+.. _configuration:
+
+Configuration Sources
+*********************
+
+Internally, cloud-init builds a single configuration that is then referenced
+throughout the life of cloud-init. The configuration is built from multiple
+sources such that if a key is defined in multiple sources, the higher priority
+source overwrites the lower priority source.
+
+Base Configuration
+==================
+
+From lowest priority to highest, configuration sources are:
+
+* **Hardcoded config**: Config_ that lives within the source of cloud-init
+  and cannot be changed.
+* **Configuration directory**: Anything defined in ``/etc/cloud/cloud.cfg`` and
+  ``/etc/cloud/cloud.cfg.d``.
+* **Runtime config**: Anything defined in ``/run/cloud-init/cloud.cfg``.
+* **Kernel command line**: On the kernel command line, anything found between
+  ``cc:`` and ``end_cc`` will be interpreted as cloud-config user data.
+
+These four sources make up the base configuration.
+
+Vendor and User Data
+====================
+Added to the base configuration are:
+
+* **Vendor data**: :ref:`Data<vendordata>` provided by the datasource
+* **User data**: :ref:`Data<user_data_formats>` also provided by
+  the datasource
+
+These get fetched from the datasource and are defined at instance launch.
+
+.. note::
+  While much of what is defined in the base configuration can be overridden by
+  vendor data and user data, base configuration sources do not conform to
+  :ref:`#cloud-config<topics/format:Cloud Config Data>`
+
+Network Configuration
+=====================
+Network configuration happens independently from other cloud-init
+configuration. It is sourced from the cloud provided metadata service and is
+not user editable. If you need to change your network configuration,
+your options are:
+
+* Use the cloud-provided tools to update your network topology in the cloud.
+* :ref:`Disable cloud-init network configuration entirely
+  <topics/network-config:Disabling Network Configuration>`.
+  This leaves network configuration entirely up to you.
+
+Specifying Configuration
+==========================
+
+End users
+---------
+Pass :ref:`user data<user_data_formats>` to the cloud provider.
+Every platform supporting cloud-init will provide a method of supplying
+user data. If you're unsure how to do this, reference the documentation
+provided by the cloud platform you're on. Additionally, there may be
+related cloud-init documentation in the :ref:`datasource<datasources>`
+section.
+
+Once an instance has been initialized, the user data may not be edited.
+It is sourced directly from the cloud, so even if you find a local file
+that contains user data, it will likely be overwritten next boot.
+
+To modify the run frequency of a particular cloud-init module,
+:ref:`cloud-init single<cli_single>` may be used. This can be used to
+disable a module you no longer with to run.
```

Comment:
This is not true is it?  The CLI doesn't modify the run frequency of a module, it allows you to override that frequency manually on individual calls.

Maybe something like:
```suggestion
To override the default run frequency of a particular cloud-init module,
:ref:`cloud-init single<cli_single>` may be invoked from scripts, cloud-config `runcmd` or `boodcmd` sections .
 ```

Not sure we want to document the following but:


To modify the default frequency of a particular cloud-init module to `once-per-instance`, `always`, or `once`. Base config in /etc/cloud/cloud.cfg(.d) is needed. To override default frequency for a particular module, provide a list of values for the specific module name of the format `[<module_name>, <frequency>]` in any of the cloud-init boot stages `cloud_init_modules`, `cloud_config_modules` or `cloud_final_modules`:

```
cloud_init_modules:
 - [ write-files, once-per-instance]
....
```

Reply:
Doh, good catch. I misread the documentation for cloud-init single, so I've also updated that documentation to be slightly clearer. I also removed this bit for reasons I'll explain in a larger comment on the PR.